url3aa01d34ec_katsuster_ememu_tgz-pJ8-net_katsuster_ememu_ui_MainJ8/src/net/katsuster/ememu/arm/core/CoProcStdv5.java

```diff
@@ -0,0 +1,397 @@
+        case CR08_UTLB_INVALL:
+            //System.out.printf("I&D-TLB  : all invalidated.\n");
+            break;
+        case CR08_UTLB_INVV:
+            System.out.printf("i&D-TLB  : invalidated 0x%08x.\n", val);
+            break;
+        case CR08_ITLB_INVALL:
+            //System.out.printf("I-TLB    : all invalidated.\n");
+            break;
+        case CR08_ITLB_INVV:
+            //System.out.printf("I-TLB    : invalidated 0x%08x.\n", val);
+            break;
+        case CR08_DTLB_INVALL:
+            //System.out.printf("D-TLB    : all invalidated.\n");
+            break;
+        case CR08_DTLB_INVV:
+            //System.out.printf("D-TLB    : invalidated 0x%08x.\n", val);
+            break;
+        default:
+            super.setCReg(cn, val);
+            break;
+        }
+    }
+
+    /**
+     * crn01: 制御ビット
+     *
+     * @param val 新たなレジスタの値
+     */
+    public void setSCTLR(int val) {
+        boolean v = BitOp.getBit32(val, 13);
+        boolean r = BitOp.getBit32(val, 9);
+        boolean s = BitOp.getBit32(val, 8);
+        boolean a = BitOp.getBit32(val, 1);
+        boolean m = BitOp.getBit32(val, 0);
+
+        //System.out.printf("SCTLR    : 0x%x.\n", val);
+        //System.out.printf("  V      : %b.\n", v);
+        //System.out.printf("  R      : %b.\n", r);
+        //System.out.printf("  S      : %b.\n", s);
+        //System.out.printf("  A      : %b.\n", a);
+        //System.out.printf("  M      : %b.\n", m);
+
+        //v: ハイベクタ、0: 無効、1: 有効
+        getCPU().setHighVector(v);
+        //r: ROM 保護ビット
+        getCPU().getMMU().setROMProtect(r);
+        //s: システム保護ビット
+        getCPU().getMMU().setSystemProtect(s);
+        //a: アドレスアラインメントチェック、0: 無効、1: 有効
+        getCPU().getMMU().setAlignmentCheck(a);
+        //m: MMU イネーブルビット、0: 無効、1: 有効
+        getCPU().getMMU().setEnable(m);
+
+        super.setCReg(CR01_MMU_SCTLR, val);
+    }
+
+    /**
+     * crn02: メモリの保護と制御、MMU 変換テーブル
+     *
+     * @param val 新たなレジスタの値
+     */
+    public void setTTBR0(int val) {
+        //int base = (val >> 14) & 0x3ffff;
+
+        //System.out.printf("TTBR0    : 0x%x.\n", val);
+        //System.out.printf("  base   : 0x%x.\n", base);
+
+        //MMU の状態を更新する
+        getCPU().getMMU().setTableBase(val);
+
+        super.setCReg(CR02_MMU_TTBR0, val);
+    }
+
+    /**
+     * crn03: メモリの保護と制御、MMU ドメインアクセス制御
+     *
+     * @param val 新たなレジスタの値
+     */
+    public void setDACR(int val) {
+        for (int i = 0; i < 16; i++) {
+            getCPU().getMMU().setDomainAccess(i, (val >> (i * 2)) & 0x3);
+        }
+
+        super.setCReg(CR03_MMU_DACR, val);
+    }
+
+    /**
+     * 割り込み待ち。
+     *
+     * @param val 新たなレジスタの値
+     */
+    public void waitInt(int val) {
+        ARMv5 cpu = getCPU();
+
+        synchronized (cpu) {
+            while (!cpu.isRaisedInterrupt()) {
+                try {
+                    cpu.wait(100000);
+                } catch (InterruptedException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)